klickhouse/src/client.rs

```diff
@@ -414,9 +414,8 @@ impl Client {
                 column_types: first_block.column_types.clone(),
                 column_data: IndexMap::new(),
             };
-            let types = first_block.column_types.values().collect::<Vec<_>>();
             rows.into_iter()
-                .map(|x| x.serialize_row(&types[..]))
+                .map(|x| x.serialize_row(&first_block.column_types))
```

Comment:
From my understanding, the order does not matter anymore for serialization.

For deserialization, it does because of the index-based field matching. In deserialization, I also noticed that we transform an IndexMap into a Vec, which is making `flattening` from #30 a bit less efficient than it could be, as we can't access fields by name in constant time (see the end of the `flatten support` section).

Although it means another breaking change later, I agree it would be better to do this in the future rather than right now.